### ISSUE TITLE
Error initializing parameters when inheriting from tf.keras.model！！！！

### ISSUE THREAD

**Issue by 1148330040** [tensorflow/tensorflow#51447]
env: tf2.2,  linux

**If there are two input parameters inherited from the tf.keras.model class and one of them is of bool type, it will directly report an error like this:**
![1628757059(1)](https://user-images.githubusercontent.com/34124260/129164925-5ca64255-cd04-45fe-978d-1cf8f4d42a27.png)

**However, the interesting thing is that if there are three input parameters and the third parameter has no effect, it will be correct，like this：**
![1628757177(1)](https://user-images.githubusercontent.com/34124260/129165214-ffe000af-e70a-46bd-bb78-2f11694680da.png)
![1628757205(1)](https://user-images.githubusercontent.com/34124260/129165314-a2ff8e99-45a0-4634-bb44-5eb6c7efb33b.png)
**If it has only one initial input, there will be no error if it is of bool type**

This is an interesting question. I don't know whether there is a problem with my environment or whether there is a small bug in the code

**Comment by sushreebarsa** [comment]
@1148330040 In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by 1148330040** [comment]
this is my complete code:

class ModelBertCrf4Sub(tf.keras.Model):
    """该类用以获取subject
    使用bert&crf模型, 以序列标注的方法进行训练
    """
    def __init__(self, output_dim, use_crf):
        super(ModelBertCrf4Sub, self).__init__(output_dim, use_crf)
        self.use_crf = use_crf
        self.bert = TFBertModel.from_pretrained('hfl/chinese-bert-wwm-ext')
        self.dropout = tf.keras.layers.Dropout(0.3)
        self.dense = tf.keras.layers.Dense(output_dim)
        self.other_params = tf.Variable(tf.random.uniform(shape=(output_dim, output_dim)))

    @tf.function
    def call(self, batch_data):
        ids, masks, tokens, subject_target = batch_data
        input_seq_len = tf.cast(tf.reduce_sum(masks, axis=1), dtype=tf.int32)
        hidden = self.bert(ids, masks, tokens)[0]
        dropout_inputs = self.dropout(hidden, 1)
        sub_predict = self.dense(dropout_inputs)
        log_likelihood = None
        if self.use_crf:
            log_likelihood, self.other_params = tfa.text.crf.crf_log_likelihood(sub_predict,
                                                                                subject_target,
                                                                                input_seq_len,
                                                                                self.other_params)
            sub_predict, _ = tfa.text.crf_decode(sub_predict, self.other_params , input_seq_len)

        return sub_predict, hidden, log_likelihood

Tf2.5 and keras versions are limited. My side is tf2.2, so please wait a minute

**Comment by 1148330040** [comment]
this is my complete code, my tf is tf2.2, tf2.5 need wait a moment，thanks！
`class ModelBertCrf4Sub(tf.keras.Model):
    """该类用以获取subject
    使用bert&crf模型, 以序列标注的方法进行训练
    """
    def __init__(self, output_dim, use_crf):
        super(ModelBertCrf4Sub, self).__init__(output_dim, use_crf)
        # 如果使用embedding需要用到此参数, input_dim=len(vocabs)
        self.use_crf = use_crf
        self.bert = TFBertModel.from_pretrained('hfl/chinese-bert-wwm-ext')
        self.dropout = tf.keras.layers.Dropout(0.3)
        self.dense = tf.keras.layers.Dense(output_dim)
        self.other_params = tf.Variable(tf.random.uniform(shape=(output_dim, output_dim)))

    @tf.function
    def call(self, batch_data):
        ids, masks, tokens, subject_target = batch_data
        input_seq_len = tf.cast(tf.reduce_sum(masks, axis=1), dtype=tf.int32)
        hidden = self.bert(ids, masks, tokens)[0]
        dropout_inputs = self.dropout(hidden, 1)
        sub_predict = self.dense(dropout_inputs)
        log_likelihood = None
        if self.use_crf:
            log_likelihood, self.other_params = tfa.text.crf.crf_log_likelihood(sub_predict,
                                                                                subject_target,
                                                                                input_seq_len,
                                                                                self.other_params)
            sub_predict, _ = tfa.text.crf_decode(sub_predict, self.other_params , input_seq_len)

        return sub_predict, hidden, log_likelihood`

**Comment by 1148330040** [comment]
> @1148330040 In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Could you please try on latest stable version of TF 2.5 and let us know if this is still an issue.Thanks!

In tf2.5 and tf2.4 environments：
Since it is not necessary to declare the two input parameters (use_crf and input_dim) in super(ModelBertCrf4Sub, self).__init__(), so the above error does not occur：

![1628761526(1)](https://user-images.githubusercontent.com/34124260/129176044-bd660108-416a-45f3-a75f-bf6ec9c30ac1.png)

**Comment by sushreebarsa** [comment]
@1148330040 Thank you for the update ! Could you please let us know if this issue is resolved for you ? Thanks!
